src/v2/client/completeLoginFlow.js

```diff
@@ -0,0 +1,50 @@
+/*!
+ * Copyright (c) 2021, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import Enums from 'util/Enums';
+import Errors from 'util/Errors';
+import { toQueryString } from '@okta/okta-auth-js';
+
+import { clearTransactionMeta } from './transactionMeta';
+
+export async function completeLoginFlow (settings, idxResponse) {
+  const { interactionCode } = idxResponse;
+  const authClient = settings.getAuthClient();
+  const transactionMeta = authClient.transactionManager.load();
+  const shouldRedirect = settings.get('mode') === 'remediation';
+
+  // Should this transaction be completed at the `redirectUri` ?
+  if (shouldRedirect) {
+    const redirectUri = settings.get('redirectUri');
+    if (!redirectUri) {
+      throw new Errors.ConfigError('"redirectUri" is required');
+    }
+    const { state } = transactionMeta;
+    const qs = toQueryString({ 'interaction_code': interactionCode, state });
+    window.location.replace(redirectUri + qs);
```

Comment:
Interesting choice to not save this to session history. For my own curiosity - why are we doing that here versus using the traditional `location.assign` or `location = ${someUrl}`?

Reply:
This is a great question. If we are on a callback route, calling `location.replace` is appropriate/optimal because it keeps the user from seeing an error state if they press `back`. Instead, they go to the page they were at before they attempted to sign in.

However, if the flow is occurring in a SPA and/or there has been no redirection, then this would be erroneously eating a history entry.

In this case, we are not handling a redirect. We are actually **forcing** a redirect where none is required. We have an interaction code and we could exchange it for tokens, but the configuration is telling us to perform a redirect anyways. The use case here is either the developer wants consistency (always redirect) or because the transaction originated on the server (web client type).

So based on this, I do think `location.assign` is the better choice and I should add some more comments around this code block.